src/www/ui/api/Controllers/MaintenanceController.php

```diff
@@ -0,0 +1,61 @@
+<?php
+/*
+ SPDX-FileCopyrightText: © 2022 Samuel Dushimimana <dushsam100@gmail.com>
+
+ SPDX-License-Identifier: GPL-2.0-only
+*/
+
+namespace Fossology\UI\Api\Controllers;
+
+
+use Fossology\Lib\Dao\UserDao;
+use Fossology\UI\Api\Helper\ResponseHelper;
+use Fossology\UI\Api\Models\Info;
+use Fossology\UI\Api\Models\InfoType;
+use maintagent;
+use Psr\Http\Message\ServerRequestInterface;
+
+/**
+ * Valid status inputs
+ */
+
+
+/**
+ * @class ReportController
+ * @brief Controller for Maintenance model
+ */
+class MaintenanceController extends RestController
+{
+  /**
+   * Create maintenance
+   *
+   * @param ServerRequestInterface $request
+   * @param ResponseHelper $response
+   * @param array $args
+   * @return ResponseHelper
+   */
+  public function createMaintenance($request, $response, $args)
+  {
+    $body = $this->getParsedBody($request);
+    $maintain  = $this->restHelper->getPlugin('maintagent');
+
+    //Remove all duplicate options
+    $body['options'] = array_unique($body['options']);
+
+    //Check if all the given keys exist in the known options array
+    $alteredOptions = array();
+    foreach ($body['options'] as $key) {
+      if (!array_key_exists($key, $maintain->getOptions())) {
```

Comment:
Please store response from `$maintain->getOptions()` in a variable to reduce function calls.